src/main.rs

```diff
@@ -78,11 +81,14 @@ fn main() -> Result<()> {
     // Try to enable BPF stats
     let fd = unsafe { bpf_enable_stats(libbpf_sys::BPF_STATS_RUN_TIME) };
     if fd < 0 {
-        return Err(anyhow!("Failed to enable BPF stats"));
+        if let Err(err) = fs::write(BPF_STATS_ENABLED, b"1") {
```

Comment:
My preferred approach would be to find a way to detect that the Kernel supports the `BPF_STATS_RUN_TIME` bpf syscall, like checking the Kernel version.

Falling back to the `BPF_STATS_ENABLED` path if the syscall fails feels brittle. The syscall may have failed for a different reason and the user should know. We should only do the fallback way if we know the user is on an older kernel.

Reply:
@zf1575192187 I pushed a change to your branch to only enable stats via procfs if the Kernel version is below 5.8

Reply:
Ok, thanks.

Reply:
Could you pull that change into your branch and push it to this PR?

Reply:
> Could you pull that change into your branch and push it to this PR?

OK, and do you have any good suggestions for app panics?

Reply:
I don't right now. But we can address that in a follow-up PR since there is already a lot of changes here. It may require refactoring the main func.